src/utils/media-utils.js

```diff
@@ -5,16 +5,21 @@ if (process.env.NODE_ENV === "development") {
   resolveMediaUrl = `https://${process.env.DEV_RETICULUM_SERVER}${resolveMediaUrl}`;
 }
 
+const farsparkCache = new Map();
 export const resolveFarsparkUrl = async url => {
   const parsedUrl = new URL(url);
-  if ((parsedUrl.protocol !== "http:" && parsedUrl.protocol !== "https:") || isHostWhitelisted(parsedUrl.hostname))
-    return { raw: url, origin: url };
+  if (farsparkCache.has(url)) return farsparkCache.get(url);
 
-  return await fetch(resolveMediaUrl, {
-    method: "POST",
-    headers: { "Content-Type": "application/json" },
-    body: JSON.stringify({ media: { url } })
-  }).then(r => r.json());
+  const resolved =
+    (parsedUrl.protocol !== "http:" && parsedUrl.protocol !== "https:") || isHostWhitelisted(parsedUrl.hostname)
+      ? { raw: url, origin: url }
```

Comment:
In this case, we do not "resolve" the media url via farspark, which makes me think `farsparkCache` and `resolveFarsparkUrl` are misnomers. Are there better names for these? 

I haven't thought of better names, but here are some suggestions:
- `resolveUrl`/`urlsCache`
- `resolvedUrlsCache` (ew)
- `resolve` for the function and `urls` for the cache
- `farsparkifyIfNecessary` (ok not this one)

Reply:
I kind of like `resolve` and `urls` (or `urlsCache`)

Reply:
But `resolve` doesn't really tell you what's happening. I think it's the 2nd definition here:

resolve: 
(verb) settle or find a solution to (a problem, dispute, or contentious matter).
(verb) decide firmly on a course of action.
(noun) firm determination to do something.

Reply:
Renamed to `resolveMedia` and `resolveMediaCache`